### DISCUSSION TITLE
Image appears as application/octet-stream

### DISCUSSION THREAD

**Discussion by miguelpruivo** [alexkn/node-red-contrib-valetudo#22] [Q&A] [answered]
Hi, I subscribed the `map-data` topic and I'm trying to decode the generated input from this package to base64 (buffer <-> base64) and then using the generated base64 to actually generate an image.

However, when trying to using online decoders, I always seem to get an invalid image because the data is in `application/octet-stream`. How can I convert it to an image?

<img width="904" alt="Captura de ecrã 2022-05-04, às 12 15 55" src="https://user-images.githubusercontent.com/27860743/166671282-06d2aec4-8dd2-4a12-83a9-243ea17fa199.png">

Thanks!

**Comment by alexkn** [comment, marked as answer]
The valetudo-map-png outputs binary data in png format as buffer.
The base64 should encode it in base64 so it can be send to dashboard template node to render it with a data url.

What do you want to do with the image?

**Reply by miguelpruivo** [reply]
I was trying to send it to telegram, but got it working without having to encode to base64.
